drivers/usb/udc/udc_dwc2_vendor_quirks.h

```diff
@@ -182,10 +183,17 @@ static inline int usbhs_enable_nrfs_service(const struct device *dev)
 static inline int usbhs_enable_core(const struct device *dev)
 {
 	NRF_USBHS_Type *wrapper = USBHS_DT_WRAPPER_REG_ADDR(0);
+	k_timeout_t timeout = K_FOREVER;
+
+	if (CONFIG_UDC_DWC2_USBHS_VBUS_READY_TIMEOUT) {
```

Comment:
```suggestion
	if (IS_ENABLED(CONFIG_UDC_DWC2_USBHS_VBUS_READY_TIMEOUT)) {
```

Reply:
I don't think it should be done with `IS_ENABLED`. It's an integer Kconfig, not a boolean one that you can enable or not.

Reply:
Does this even compile when `depends on NRFS_HAS_VBUS_DETECTOR_SERVICE` is not fulfilled?

Reply:
Oh, I haven't thought about that case. Maybe it should be `if (IS_ENABLED(NRFS_HAS_VBUS_DETECTOR_SERVICE) && CONFIG_UDC_DWC2_USBHS_VBUS_READY_TIMEOUT)`?

Reply:
What will `CONFIG_UDC_DWC2_USBHS_VBUS_READY_TIMEOUT` evaluate to (if anything) when `NRFS_HAS_VBUS_DETECTOR_SERVICE` is not set?

Reply:
Correction - it doesn't work with `IS_ENABLED`. `IS_ENABLED(x)` evaluates to true only if the `x` is defined to 1. So it seems that the only solution is to use `#if NRFS_HAS_VBUS_DETECTOR_SERVICE`.